src/lib.rs

```diff
@@ -3908,6 +3965,80 @@ mod tests {
         }
     }
 
+    mod for_dst {
```

Comment:
What's the thinking behind having these in a nested module as opposed to just naming them `test_dst_layout_for_sized` etc?

Reply:
- reduces repetition in the names
- makes it possible to code-collapse the entire group in an editor

My preference (not in this PR) would be even to go a little further and drop `test_` from all tests. All of the tests are already located in a module called "tests".